### PR TITLE
qgis: add 3.36 and 3.40, fix proj depend

### PR DESCRIPTION
**Pull request by Chrismarsh**
QGis 3.34.4 does not have the backport of the [fix](https://github.com/qgis/QGIS/commit/d6493f7f00d8920ac1ea103b186d44fb4972bc54) for building against proj 9.4+ -- only [3.34.5+](https://github.com/qgis/QGIS/compare/final-3_34_4%5E...final-3_34_5) does. This fix ensures a compatible proj when building `@:3.34.4`.

In addition, this adds the most recent release (3.40.1), prefers the newest LTR (3.34.13), and deprecates the (apparently? I didn't test) not building 3.36 version.

### CODE REVIEW

**Comment by Sinan81** [comment]
@Chrismarsh thanks for improving the spack package file for QGIS! I'll try to build it from scratch and approve it afterwards. 

It would be useful for me if you could indicate the details of the build like: `spack spec /abcde..` where `/abcde..` is the build hash which can be obtained via `spack find -l qgis` as you might already know. What Spack version did you use? No hurry. Thanks

File: var/spack/repos/builtin/packages/qgis/package.py [hunk 1]
```diff
@@ -18,15 +18,20 @@ class Qgis(CMakePackage):
...
-
-    # TODO version 3.36 isn't building right now.
-    version("3.36.0", sha256="1b64bc92660bf07edc6b6478fc6a13656149e87d92eabe5c3db9493072506e2c")
+    version("3.40.1", sha256="53110464c9f5ba5562c437e1563ab36dad2f218e6e7d1c0cfbe5b6effe241c8e")
+    #  version 3.36 isn't building right now.
+    version(
+        "3.36.0",
+        sha256="1b64bc92660bf07edc6b6478fc6a13656149e87d92eabe5c3db9493072506e2c",
+        deprecated=True,
+    )
```
Comment (Sinan81) [inline var/spack/repos/builtin/packages/qgis/package.py:27]: ~~version 3.36 can be removed since it's not an LTR.~~ NVM I just saw that you deprecated this version.

**Comment by Sinan81** [comment]
qgis@3.34.13. builds fine. ran into an unrelated minor issue with libspatialite & libxml2. will submit an MR soon. @adamjstewart please approve.

File: var/spack/repos/builtin/packages/qgis/package.py [hunk 2]
```diff
@@ -127,6 +132,9 @@ class Qgis(CMakePackage):
     depends_on("proj@4.9.3:", when="@3.8.2:")
     depends_on("proj@7.2:", when="@3.28:")
     depends_on("proj@:8", when="@3.28")  # build fails with proj@9
+    # fails to build with proj 9.4+ until the backported patch in 3.34.5
+    # https://github.com/qgis/QGIS/compare/final-3_34_4%5E...final-3_34_5
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/qgis/package.py]: It's a bit hard to parse this long list of commits, I would personally link to one or both of:

* https://github.com/qgis/QGIS/pull/56761
* https://github.com/qgis/QGIS/pull/56765

Comment (Chrismarsh) [reply var/spack/repos/builtin/packages/qgis/package.py]: It took me a while to dig through what release this was in, but noting the release and then linking to https://github.com/qgis/QGIS/pull/56761 makes sense to me

**Comment by Chrismarsh** [comment]
@Sinan81 
```
spack debug report
* **Spack:** 0.23.0.dev0 (11aa02b37adcb04925c75e8746587ea65b5cd812)
* **Python:** 3.9.20
* **Platform:** linux-centos9-zen2
```

What I tested
```
qgis@3.40.1%gcc@14.1.0~3d+analysis~apidoc~astyle+bindings~clang_tidy+core~custom_widgets+desktop+georeferencer~globe~grass7+gui+internal_mdal+internal_o2~ipo+oauth2_plugin~oracle~pdal+postgresql~py_compile+qsciapi~qspatialite+qt5serialport~qtmobility~qtwebkit~quick~qwtpolar~server+staged_plugins+thread_local~txt2tags build_system=cmake build_type=Release generator=make arch=linux-centos9-zen2

```

**Review by adamjstewart — APPROVED** [review]
